### PR TITLE
Package the Messaging java file and aar separately

### PR DESCRIPTION
**Pull request by a-maurice**
Instead of baking the Messaging java file and C++ aar into the srcaar file, keep them separate. This is to support Unity 2023, which has modified the behavior of UnityPlayerActivity, specifically the mUnityPlayer member variable.  Note that additional work is still needed for the new UnityPlayerGameActivity logic.

Depends on https://github.com/firebase/firebase-cpp-sdk/pull/1205
***

Tested locally with 2019.4.40f1, 2021.3.13f1, and 2023.1.0b3.
https://github.com/firebase/firebase-unity-sdk/actions/runs/4180060512
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [x] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: cmake/android_dependencies.cmake [hunk 1]
```diff
@@ -65,6 +65,7 @@ set(FIREBASE_MESSAGING_ANDROID_DEPS
     "com.google.firebase:firebase-messaging:23.1.1"
     "com.google.firebase:firebase-analytics:21.2.0"
     "com.google.firebase:firebase-iid:21.1.0"
+    "com.google.flatbuffers:flatbuffers-java:1.12.0"
```
Comment (cynthiajoan) [inline cmake/android_dependencies.cmake:68]: Are we going to update the version if flatbuffer has a new release?

File: cmake/unity_pack.cmake [hunk 2]
```diff
@@ -308,19 +309,26 @@ endfunction()
...
   if("${UNITY_PACK_PACK_PATH}" STREQUAL "")
     set(UNITY_PACK_PACK_PATH ${UNITY_PACK_DEFAULT_CS_PATH})
   endif()
 
-  # meta file
-  install(
-    FILES ${file}
-    DESTINATION ${UNITY_PACK_PACK_PATH}
-  )
+  if ("${UNITY_PACK_RENAME_TO}" STREQUAL "")
```
Comment (cynthiajoan) [inline cmake/unity_pack.cmake:320]: Can we add some comments about this section?

Comment (a-maurice) [reply cmake/unity_pack.cmake:322]: Added some more comments.

File: cmake/android_dependencies.cmake (hunk 1 above)
Comment (a-maurice) [reply cmake/android_dependencies.cmake:68]: We'll keep this in sync with whatever version the C++ repo is using.

**Review by cynthiajoan — APPROVED** [review]

**Pull request merged**